### ISSUE TITLE
uncaught Object: Invalid encoding: FE512M1.52631043660800200000000000000000000

### ISSUE THREAD

**Issue by evbogue** [ssbc/scuttlebot#502]
I'm also getting this error in my client now. I thought maybe it's a new Node.js version thing, but I backed up to LTS and I get the same error.

I get this in Patchless and Decent.

Anyone else seeing this?

```
message
:
"Invalid encoding: FE512M1.52631043660800200000000000000000000"
name
:
"EncodingError"
stack
:
"EncodingError: Invalid encoding: FE512M1.52631043660800200000000000000000000↵    at /home/ev/decent/node_modules/level-iterator-stream/index.js:36:28↵    at /home/ev/decent/node_modules/leveldown/node_modules/abstract-leveldown/abstract-iterator.js:24:16↵    at /home/ev/decent/node_modules/leveldown/iterator.js:43:7↵    at _combinedTickCallback (internal/process/next_tick.js:131:7)↵    at process._tickCallback (internal/process/next_tick.js:180:9)"
```

**Comment by evbogue** [comment]
@staltz yah, I'm completely bricked over here right now. Thought my altnet is working just fine. 

Right now I'm not getting this error though, instead I'm just getting nothing anywhere. Nothing is kind of hard to debug! :cry:

**Comment by evbogue** [comment]
I should mention that I'm on cutting-edge everything. Latest Node, npm, Arch, Chrome, Firefox, Scuttlebot. 

@gwenbell's instance is also bricked, in fact it bricked first, so that is why I ran into this.

**Comment by evbogue** [comment]
just installing secure-scuttlebutt@18.0.5 did not fix the main ssb network for me. 

But I should note I'm not getting that above error anymore, so maybe it did solve the above issue.

Let me try regenerating flume...

**Comment by staltz** [comment]
@dominictarr that sounds like a good explanation, but maybe it doesn't cover the issue I had (I'll try to later create a good reproduction of it and report on flumeview-level), because I was running tests with scuttle-testbot which basically starts with a fresh db

**Comment by gwenbell** [comment]
This started a few days ago with a pacman -Syu. I thought it might have been a new version of Node, but now that I'm seeing this thread, I don't think it's just one thing. This is what I'm seeing this morning after doing an rm -rf of node_modules and then an npm install:
```
make: *** [leveldown.target.mk:113: Release/obj.target/leveldown/src/batch.o] Error 1
make: Leaving directory '/home/g/Decent/node_modules/leveldown/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:235:12)
gyp ERR! System Linux 4.16.8-1-ARCH
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/g/Decent/node_modules/leveldown
gyp ERR! node -v v10.1.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm WARN decent-ssb@4.4.1 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! Maximum call stack size exceeded
```

**Comment by evbogue** [comment]
Update from tonight: @gwenbell is still getting `call stack size exceeded`. I'm able to install node modules, but the main ssb network doesn't work for me without errors. I am able to run the decent network, without errors. @gwenbell is unable to run either, as node modules aren't installing.

I'm learning towards the two-versions-of-level theory, as @gwenbell started to see an advert for leveldb's opencollective page around the same time as this error.

**Comment by evbogue** [comment]
ok, I got the main network up and running again, on my own computer at least.

I did three things:

1. I installed the latest `secure-scuttlebutt`. 
2. I regenerated the flumeviews by `rm -rf`ing `.ssb/flume`
3. I ran a `pacman -Syu` and got a new version of python.

I'm not sure which one of these things fixed the issue, but it didn't start working again until item 3. So it might have been that Arch Linux shipped a dud version of python and that was tripping up the leveldb build?

**Comment by dominictarr** [comment]
@evbogue what node version are you on? there should be builds of leveldb and that shouldn't automatically build. I'd recommend managing node with nvm not using pacman's node version.

**Comment by evbogue** [comment]
@dominictarr Most of the time I use the node package in Arch Linux, which trends towards being the latest version. 

When this error occurred I switched to using `nvm` and tried switching between node@lts and node@10.1.0. 

The weird thing about this issue is leveldb _was_ building when the error occurred, but now when I use `npm install` it appears to be using the leveldb pre-build. So, for some reason `npm` was building leveldb even though it probably didn't need to? 

Anyway, this issue is resolved for both @gwenbell and I. I'm fairly sure if no one else sees this issue, then this issue is Arch Linux specific and was resolved within the Arch Linux ecosystem.
